### PR TITLE
notifications without shelling out

### PR DESCRIPTION
**Pull request by hoodie**
Hi there, awesome tool.
I saw that you check for local tools like notify-send and osascript to produce notifications. There is actually already a crate for that :D 
This is what it would look like.

cheers

### CODE REVIEW

**Comment by greyblake** [comment]
@hoodie Hey, thanks for the pull request!
I've seen that crate, but I was under impression that it does not support Mac, the description says `Send Desktop Notifications on linux and bsd`.

Hm..By some reason the PR breaks the build.

**Comment by greyblake** [comment]
Tests pass locally for me. 
Looks like `notify-rust` uses dbus

**Comment by greyblake** [comment]
@hoodie  The version with `notify-rust` shows only `title` without message:

image

File: src/reactor.rs [hunk 1]
```diff
@@ -114,7 +116,15 @@ impl Reactor {
                 let stderr_output = stderr_buffer.lock().unwrap().clone();
 
                 let report = self.report_builder.identify(exit_status.success(), &stdout_output, &stderr_output);
-                self.config.notifier.notify(report);
+                let icon = match report.outcome {
+                    Outcome::TestsPassed => "face-angel",
+                    Outcome::TestsFailed | Outcome::CompileError => "face-angry"
+                };
+                Notification::new()
+                    .summary(report.title())
```
Comment (greyblake) [inline src/reactor.rs]: Looks like we need to set `body` here as well, when `report.detail` is `Some(String)`

**Review by greyblake — CHANGES_REQUESTED** [review]

**Comment by hoodie** [comment]
sorry, I didn't check the CI on this PR - I just added the required config to the .travis.yml

**Comment by greyblake** [comment]
@hoodie  Thank you!)

**Pull request merged**